baseplate/secrets/fetcher.py

```diff
@@ -18,7 +20,7 @@
         secret/three,
 
 where each secret is a path to look up in Vault. The daemon authenticates with
-Vault as the EC2 server it is running on. Vault can map that context to
+Vault using the auth backend designated by `auth_type`. Vault can map that context to
```

Comment:
there's a quick note of auth type above, but no list of the auth types with prose explanations of their purpose / configurables. it'd be nice to add that to this module docstring.